ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -0,0 +1,736 @@
+                series.options.link && series.options.link.lineWidth
+            ),
+            attribs = Series.prototype.pointAttribs.call(series, point, state);
+
+        if (point.isLink) {
+            attribs.stroke = linkColor;
+            attribs['stroke-width'] = linkLineWidth;
+            delete attribs.fill;
+        } else {
+            if (borderRadius) {
+                attribs.r = borderRadius;
+            }
+        }
+        return attribs;
+    }
+
+    public drawPoints(): void {
+        super.drawPoints.apply(this, arguments);
+        const points = this.points;
+        this.points = this.links as any;
+        ColumnSeries.prototype.drawPoints.apply(this, arguments);
+        this.points = points;
+    }
+    /**
+     * Run translation operations for one node.
+     * @private
+     */
+    public translateNode(point: TreegraphPoint): void {
+        const chart = this.chart,
+            node = point.node,
+            plotSizeY = chart.plotSizeY as number,
+            plotSizeX = chart.plotSizeX as number,
+            // Get the layout modifiers which are common for all nodes.
+            { ax, bx, ay, by } = this.layoutModifier,
+            x = ax * node.xPosition + bx,
+            y = ay * node.yPosition + by,
+            level = this.mapOptionsToLevel[node.level] || {},
+            markerOptions = merge(
+                this.options.marker,
+                level.marker,
+                point.options.marker
+            ),
+            symbol = markerOptions.symbol,
+            height = node.nodeSizeY,
+            width = node.nodeSizeX,
+            reversed = this.options.reversed,
+            nodeX = node.x = (chart.inverted ?
+                plotSizeX - width / 2 - x :
+                x - width / 2),
+            nodeY = node.y = (!reversed ?
+                plotSizeY - y - height / 2 :
+                y - height / 2);
+
+        point.shapeType = 'path';
+        point.plotX = nodeX;
+        point.plotY = nodeY;
+        point.shapeArgs = {
+            d: symbols[symbol || 'circle'](nodeX, nodeY, width, height),
+            x: nodeX,
+            y: nodeY,
+            width,
+            height,
+            cursor: !point.node.isLeaf ? 'pointer' : 'default'
+        };
+
+        // Set the anchor position for tooltip.
+        point.tooltipPos = chart.inverted ?
+            [plotSizeY - nodeY - height / 2, plotSizeX - nodeX - width / 2] :
+            [nodeX + width / 2, nodeY];
+    }
+}
+
+/* *
+ *
+ *  Prototype Properties
+ *
+ * */
+
+interface TreegraphSeries {
+    inverted?: boolean;
+    pointClass: typeof TreegraphPoint;
+    NodeClass: typeof TreegraphNode;
+    LinkClass: typeof TreegraphLink;
+
+}
+
+namespace TreegraphSeries {
+
+}
+/* *
+ *
+ *  Registry
+ *
+ * */
+
+TreegraphSeries.prototype.pointClass = TreegraphPoint;
+TreegraphSeries.prototype.NodeClass = TreegraphNode;
+TreegraphSeries.prototype.LinkClass = TreegraphLink;
```

Comment:
This should come directly after the `interface TreegraphSeries` and you should consider use of `Utilities.extend` to save some code.